i6_models/parts/mha.py

```diff
@@ -0,0 +1,100 @@
+__all__ = ["MultiheadAttentionV1", "MultiheadAttentionV1Config", "MHARef"]
+
+import math
+from dataclasses import dataclass
+import torch
+
+from i6_models.config import ModelConfiguration
+from i6_models.util import compat
+from dataclasses import dataclass
+
+
+@dataclass
+class MultiheadAttentionV1Config(ModelConfiguration):
+    """
+    Attributes:
+        input_dim: input dim and total dimension for query/key and value projections, should be divisible by `num_att_heads`
+        num_att_heads: number of attention heads
+        att_weights_dropout: attention weights dropout
+        dropout: multi-headed self attention output dropout
+    """
+
+    input_dim: int
+    num_att_heads: int
+    att_weights_dropout: float
+    dropout: float
+
+    def __post_init__(self) -> None:
+        super().__post_init__()
+        assert self.input_dim % self.num_att_heads == 0, "input_dim must be divisible by num_att_heads"
+
+
+class MultiheadAttentionV1(torch.nn.Module):
+    """
+    Native Multihead Attention implementation based on 'Attention Is All You Need'
+    """
+
+    def __init__(self, cfg: MultiheadAttentionV1Config):
+        super().__init__()
+        self.cfg = cfg
+        self.num_att_heads = cfg.num_att_heads
+        self.input_dim = cfg.input_dim
+        self.dim_heads = self.input_dim // self.num_att_heads
+        self.layernorm = torch.nn.LayerNorm(cfg.input_dim)
+
+        self.out_proj = torch.nn.Linear(in_features=cfg.input_dim, out_features=cfg.input_dim, bias=True)
+        self.in_proj = torch.nn.Linear(in_features=cfg.input_dim, out_features=3 * cfg.input_dim, bias=True)
+
+        self.norm = math.sqrt(float(self.input_dim / self.num_att_heads))
+        self.softmax = torch.nn.Softmax(-1)
+        self.dropout = torch.nn.Dropout(cfg.att_weights_dropout)
+
+    def forward(self, query: torch.Tensor, key: torch.Tensor, value: torch.Tensor, key_padding_mask: torch.Tensor):
```

Comment:
only one parameter then instead of three, not sure what a good name is though

Reply:
> not sure what a good name is though

I propose to have `qkv` in the name.. `qkv_tensor`? Or simply `input` or `x` if we want to be boring.

Reply:
we could also do `qkv_input` ;)

On a serious note, I think maybe even `qkv` is fine, but I don't have a strong opinion on this.